### PR TITLE
throw fatal if to fault state while going to operation enabled

### LINKED ISSUE #145: Throw error when stuck in 'Waiting for Operation Enabled'
Sometimes the IMC goes to fault state when waiting for the final transition to 'Operation Enabled'. This can be seen from a (sometimes called 'bit flip') change in the printed status word.
It would save time if the fault state would be detected (through the status word change) and an error is thrown with an appropriate message.

### PR DESCRIPTION
**Pull request by martijnvandermarel**
closes #145 

haven't tested this on the exoskeleton yet.

### CODE REVIEW

File: march_hardware/src/IMotionCube.cpp [hunk 1]
```diff
@@ -467,6 +467,12 @@ bool IMotionCube::goToTargetState(IMotionCubeTargetState targetState)
     this->setControlWord(targetState.getControlWord());
     ROS_INFO_THROTTLE(0.5, "\tWaiting for '%s': %s", targetState.getDescription().c_str(),
                       std::bitset<16>(this->getStatusWord()).to_string().c_str());
+    if (targetState.getDescription() == "Operation Enabled" && this->getState(this->getStatusWord()) == IMCState::fault)
```
Comment (Ishadijcks) [inline march_hardware/src/IMotionCube.cpp]: `targetState.getDescription() == "Operation Enabled"`
should be performed with an enum although I don't understand why it is needed at all?

**Review by Ishadijcks — DISMISSED** [review]

File: march_hardware/src/IMotionCube.cpp (hunk 1 above)
Comment (martijnvandermarel) [reply march_hardware/src/IMotionCube.cpp]: I tried comparing with the enum but wasn't allowed to by the compiler for some reason. Will check later what error message was provided.

Reason why it is necessary at all is that the very first state transition in the goToOperationEnabled is from fault -> switch on disabled. Without the targetState = Operation Enabled check, an error would be thrown while it shouldn't be.

Comment (martijnvandermarel) [reply march_hardware/src/IMotionCube.cpp]: What I tried: targetState == IMotionCubeTargetState::OPERATION_ENABLED
What compiler tells me: "error: no match for ‘operator==’ (operand types are ‘march4cpp::IMotionCubeTargetState’ and ‘const march4cpp::IMotionCubeTargetState’)"
Tried googling but couldn't really find a good fix, so then went with the next best thing which was comparing with the string.

**Comment by Ishadijcks** [comment]
I think I have implemented the fix, can you try if it works?

**Comment by martijnvandermarel** [comment]
Tested on exo, works. @Ishadijcks can you re-review?

**Review by TimBuckers — APPROVED** [review]

**Pull request merged**